pages/directory/index.tsx

```diff
@@ -75,6 +77,31 @@ const Directory: NextPage<{ users: ClientUser[]; totalPages: number; error: stri
     lastCardObserver.observe(lastCardRef.current);
   }, [userData]);
 
+  const getShortenedBio = (bio:string, maxLength = 200) => {
+    if (bio.length < maxLength) return bio;
+    let i = maxLength;
+    while (bio[i] !== ' ' && i > 0) i--;
+    // if the first word is longer than the permissible length, what should be done?
+    if (i === 0) return bio.slice(0, maxLength) + '...';
+    else return bio.slice(0, i) + '...';
+  };
```

Comment:
```text
nits: [
    :80, There should be a space before the types in the function signature.
    :84, The comment asks a question that should be resolved before commit.
    :85, 86 The return on :85 makes the else keyword on :86 unnecessary.
    :85, 86 We should be using &hellip; instead of ...
]
```

That aside, is there a reason we're doing this in JavaScript instead of limiting the size of the container and using CSS' `text-overflow`?

Reply:
This was a draft implementation, I agree that swapping it with a CSS implementation would be a good idea.